### ISSUE TITLE
'eval { use B::Deparse }' is wrong [rt.cpan.org #18793]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#60]
https://rt.cpan.org/Ticket/Display.html?id=18793

```
'use' is executed at compile time and is not affected by eval.
You should use 'eval { require ... }' instead.

--- YAML-0.58/lib/YAML/Types.pm-        2006-02-02 07:04:03 +0000
+++ YAML-0.58/lib/YAML/Types.pm 2006-04-20 02:41:19 +0000
@@ -117,7 +117,7 @@ sub yaml_dump {
     }
     else {
         bless $value, "CODE" if $class;
-        eval { use B::Deparse };
+        eval { require B::Deparse };
         return if $@;
         my $deparse = B::Deparse->new();
         eval {


-- 
Alexey Tourbin
ALT Linux Team
```

**Comment by perlpunk** [comment]
can be fixed by #165

**Comment by perlpunk** [comment]
released 1.20
